sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -849,7 +853,7 @@ def get_job_logs(
                             if logline is None:
                                 finished[index] = True
                                 break
-                            logging.info("[Pod %s]: %s", pods[index], logline)
+                            print(f"[Pod {pods[index]}]: {logline}")
```

Comment:
Thank you for updating the logging.
@droctothorpe Do we want to use `print` here instead of logging ?
Maybe we should keep our logging in SDK consistent ?
For example, KFP uses `logging.info` also: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L470
And they also setup logging here: https://github.com/kubeflow/pipelines/blob/a9279843946183429f6572516acee6523de36e53/sdk/python/kfp/cli/__main__.py#L23

Reply:
Thanks, @andreyvelich! I'm still wrapping my head around some of these nuances so take these answers with a grain of salt (Python logging is surprisingly complex). 

---

> Do we want to use print here instead of logging ?

`print` is more performant and recommended by the Python docs:

<img width="803" alt="image" src="https://github.com/kubeflow/training-operator/assets/24783969/f1241ce2-9e02-4b18-98ec-d5ca1b86eb67">

Source: https://docs.python.org/3/howto/logging.html#when-to-use-logging

---

> And they also setup logging here:
> https://github.com/kubeflow/pipelines/blob/a9279843946183429f6572516acee6523de36e53/sdk/python/kfp/cli/__main__.py#L23

Note that they only do this in the CLI. That's because the CLI is a user-facing API so to speak, i.e. it's not meant to be consumed by other libraries that have their own opinions on how logging should be configured. From the Python docs:
> It is strongly advised that you do not add any handlers other than [NullHandler](https://docs.python.org/3/library/logging.handlers.html#logging.NullHandler) to your library’s loggers. This is because the configuration of handlers is the prerogative of the application developer who uses your library. The application developer knows their target audience and what handlers are most appropriate for their application: if you add handlers ‘under the hood’, you might well interfere with their ability to carry out unit tests and deliver logs which suit their requirements.

Source: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

Reply:
I think, that makes sense, thanks for sharing these links!
I agree, that we should have appropriate `logger` for the SDK (e.g. `logging.getLogger(__name__)`) to not use `root` logger, but I still think that we should use logging to log some data for the user.

The problem with `print` is that user can't identify from which library the output was generated, but with logging we can configure it.

What do you think about this:
- We can introduce a new parameter to `TrainingClient()` called `verbose`, and we can configure this parameter for various levels of logging: https://docs.python.org/3/library/logging.html#levels. We can start with (`INFO`, `WARNING`, and `ERROR`).
- Depends on this parameter we can configure our logger in the constructor. For example, for `verbose=10`
```py
self.logger = logging.getLogger(__name__)
if verbose == 10:
  self.logger.setLevel(logging.INFO)
```

- And then we are going to use `self.logger.info()` or `self.logger.warning()` or `self.logger.error()` when it is required to print some data.

- If user doesn't want to see any logs, they can always override it as follows:
```py
logger = logging.getLogger("kubeflow.training.api.training_client")
logger.setLevel(logging.NOTSET)
```
Or provide `verbose=0` as `TrainingClient()`  parameter:
```py
client = TrainingClient(verbose=0)
```

WDYT @droctothorpe @johnugeorge @kuizhiqing ?